network-api/networkapi/wagtailpages/pagemodels/products.py

```diff
@@ -1342,7 +1350,10 @@ def categories_page(self, request, slug):
                 self.cutoff_date,
                 authenticated,
                 key,
-                ProductPage.objects.filter(product_categories__category__in=[category])
+                ProductPage.objects.filter(
+                    product_categories__category__in=[category],
+                    locale=Locale.objects.get(language_code=language_code)
+                )
```

Comment:
we're doing this a few times, it might be worth investigation the possibility of `ProductPage.objects.locale(language_code).filter(...)`, or at the very least create a prefilter function along the lines of:

```python
def get_localized(ModelType, request):
    language_code = ... or settings.LANGUAGE_CODE
    return ModelType.objects.filter(locale=Locale.objects.get(language_code=language_code))
```

so that this code becomes:
```python
products = get_product_subset(
    ...,
    get_localized(ProductPage, request).filter(product_categories__category__in=[category])
)
```